Added Support for custom labels and annotations in pods created by chart

SargamPuram opened the pull request:
<!-- Please use this template for your pull request. -->
<!-- Please use the sections that you need and delete other sections -->

## This PR
<!-- add the description of the PR here -->

This PR introduces support for custom labels and annotations for Kubernetes resources controllerManager and kubeRbacProxy to enhance organization and metadata management within the cluster.



### Related Issues
<!-- add here the GitHub issue that this PR resolves if applicable -->

Fixes #677 

### Notes
<!-- any additional notes for this PR -->

### Follow-up Tasks
<!-- anything that is related to this PR but not done here should be noted under this section -->
<!-- if there is a need for a new issue, please link it here -->

### How to test
<!-- if applicable, add testing instructions under this section -->

toddbaert commented:
Hey @SargamPuram ! @cpitstick-latai started work on this as well. Check it [here](https://github.com/open-feature/open-feature-operator/pull/681). What he has done fixes the errors in your build. The problem is that kustomize/helm don't play nice together, so we have to have a hacky script to do some pre-processing. His implementation adds the labels and annotations to the helm-installed pods, like you did here, **but not the runtime injected ones.**

If you'd still like to help with this issue, I'd recommend that you base of https://github.com/open-feature/open-feature-operator/pull/681, and add the annotations/labels at runtime (in Go code). You can see something similar here that I did for image pull secrets: https://github.com/open-feature/open-feature-operator/pull/655 (you can see in that PR I added an additional arg to the go application/container which adds the imagePullSecrets to the injected deployment template...you'd have to do something similar for pods/labels). Perhaps you can create a PR to @cpitstick-latai 's branch?

SargamPuram commented:
> Hey @SargamPuram ! @cpitstick-latai started work on this as well. Check it [here](https://github.com/open-feature/open-feature-operator/pull/681). What he has done fixes the errors in your build. The problem is that kustomize/helm don't play nice together, so we have to have a hacky script to do some pre-processing. His implementation adds the labels and annotations to the helm-installed pods, like you did here, **but not the runtime injected ones.**
> 
> If you'd still like to help with this issue, I'd recommend that you base of #681, and add the annotations/labels at runtime (in Go code). You can see something similar here that I did for image pull secrets: #655 (you can see in that PR I added an additional arg to the go application/container which adds the imagePullSecrets to the injected deployment template...you'd have to do something similar for pods/labels). Perhaps you can create a PR to @cpitstick-latai 's branch? Or you could base on his changes in a new PR (I don't think he will mind).



> Hey @SargamPuram ! @cpitstick-latai started work on this as well. Check it [here](https://github.com/open-feature/open-feature-operator/pull/681). What he has done fixes the errors in your build. The problem is that kustomize/helm don't play nice together, so we have to have a hacky script to do some pre-processing. His implementation adds the labels and annotations to the helm-installed pods, like you did here, **but not the runtime injected ones.**
> 
> If you'd still like to help with this issue, I'd recommend that you base of #681, and add the annotations/labels at runtime (in Go code). You can see something similar here that I did for image pull secrets: #655 (you can see in that PR I added an additional arg to the go application/container which adds the imagePullSecrets to the injected deployment template...you'd have to do something similar for pods/labels). Perhaps you can create a PR to @cpitstick-latai 's branch? Or you could base on his changes in a new PR (I don't think he will mind).

Hey @toddbaert ,

Thanks for the update! I checked out your work and it looks great. I understand the challenges with kustomize and helm, and the need for a pre-processing script makes sense.

I’ll base my work off #681 and add the annotations/labels at runtime in the Go code, similar to what you did in #655 for image pull secrets. I’ll create a PR to your branch to include these changes.
Thanks!

toddbaert commented:
> > Hey @SargamPuram ! @cpitstick-latai started work on this as well. Check it [here](https://github.com/open-feature/open-feature-operator/pull/681). What he has done fixes the errors in your build. The problem is that kustomize/helm don't play nice together, so we have to have a hacky script to do some pre-processing. His implementation adds the labels and annotations to the helm-installed pods, like you did here, **but not the runtime injected ones.**
> > If you'd still like to help with this issue, I'd recommend that you base of #681, and add the annotations/labels at runtime (in Go code). You can see something similar here that I did for image pull secrets: #655 (you can see in that PR I added an additional arg to the go application/container which adds the imagePullSecrets to the injected deployment template...you'd have to do something similar for pods/labels). Perhaps you can create a PR to @cpitstick-latai 's branch? Or you could base on his changes in a new PR (I don't think he will mind).
> 
> > Hey @SargamPuram ! @cpitstick-latai started work on this as well. Check it [here](https://github.com/open-feature/open-feature-operator/pull/681). What he has done fixes the errors in your build. The problem is that kustomize/helm don't play nice together, so we have to have a hacky script to do some pre-processing. His implementation adds the labels and annotations to the helm-installed pods, like you did here, **but not the runtime injected ones.**
> > If you'd still like to help with this issue, I'd recommend that you base of #681, and add the annotations/labels at runtime (in Go code). You can see something similar here that I did for image pull secrets: #655 (you can see in that PR I added an additional arg to the go application/container which adds the imagePullSecrets to the injected deployment template...you'd have to do something similar for pods/labels). Perhaps you can create a PR to @cpitstick-latai 's branch? Or you could base on his changes in a new PR (I don't think he will mind).
> 
> Hey @toddbaert ,
> 
> Thanks for the update! I checked out your work and it looks great. I understand the challenges with kustomize and helm, and the need for a pre-processing script makes sense.
> 
> I’ll base my work off #681 and add the annotations/labels at runtime in the Go code, similar to what you did in #655 for image pull secrets. I’ll create a PR to your branch to include these changes. Thanks!

Thanks so much @SargamPuram !

FYI @cpitstick-latai

cpitstick-latai commented:
I really needed this to move forward so I think I finished https://github.com/open-feature/open-feature-operator/pull/681, please take a look.

SargamPuram commented:
> Closing as completed with #681. @SargamPuram thanks, you helped us get this done.

Hey @toddbaert and @cpitstick-latai,

Thanks so much for your support and guidance throughout this process. I appreciate the opportunity to contribute and am glad that the work was completed successfully.

While I didn't contribute as much this time, I'm eager to make more valuable contributions in the future. Looking forward to collaborating more closely and helping the project in any way I can.
